internal/charmstore/store.go

```diff
@@ -270,7 +289,7 @@ func (s *Store) insertEntity(entity *mongodoc.Entity) (err error) {
 			Read:  readPerm,
 			Write: writePerm,
 		},
-		Promulgated: entity.PromulgatedURL != nil,
+		Promulgated: promulgated,
```

Comment:
I'm almost tempted to suggest something like:

    type IntBool bool
    func (b IntBool) GetBSON() (interface{}, error) { 
        if b {
            return 1, nil
        }
        return 0, nil
    }
    func (b *IntBool) SetBSON(raw bson.Raw) error {
         var x int
         if err := raw.Unmarshal(&x); err != nil { 
            etc
    }

So we can use normal conditions in the Go code.